### ISSUE TITLE
TreeDropdownField should use page icons and indicate draft status

### ISSUE THREAD

**Issue by chillu** [silverstripe/silverstripe-admin#232]
# Overview

Page icons and status are fairly important data points to get context on more complex sites. Otherwise authors might accidentally link to pages which aren't published yet. It's common to add custom icons to pages in SilverStripe, which can be a nice replacement for showing page types

# Acceptance Criteria

 * When a tree form field is used with records supporting icons (e.g. pages), these icons show in the list/search views. They don't need to show for the selected record(s)
 * The tree form field indicates when pages aren't published yet ("DRAFT")
 * Developers can choose which status flags to show. For example, "modified" shouldn't show by default, but custom status icons like "review required" might be more important
 * Status icons leave enough room to identify a page by title

**Comment by chillu** [comment]
@newleeland @clarkepaul this needs design (medium priority)

**Comment by clarkepaul** [comment]
Cheers added to backlog @chillu

**Comment by jamessilverstripe** [comment]
![add a page - dropdown](https://user-images.githubusercontent.com/29618750/30307748-9444f414-97d4-11e7-9d5c-78db92ba9594.png)
![mobile dropdown](https://user-images.githubusercontent.com/29618750/30307750-98a27108-97d4-11e7-80f5-9e401a3a877a.png)

**Comment by chillu** [comment]
Talked with James about this yesterday, it's looking great! Note that I don't think we should show the MODIFIED flag here, since that's kind of irrelevant in this context (hence the AC about "choose which status flags to show"). The point is that it'll have to work with multiple status flags at the same time (e.g. "draft" and a custom "review required")

**Comment by clarkepaul** [comment]
@chillu You might want to reconsider keeping Modified if you are showing Draft, as you might be linking to a content on a page which isn't published yet. I think we can look at the other custom status as a separate issue as they seem like they would be module specific or non-existent as yet?

**Comment by kinglozzer** [comment]
I’m working on a site with a large number of pages and it’s currently quite difficult to navigate via the `TreeDropdownField`, even when you’re already familiar with the site structure. The visual cues from icons would definitely help

I’d love to take this on, but I fear it’s outside of my limited React knowledge. Presumably this is also partly blocked by the work needed to update the to the new page icons from https://github.com/silverstripe/silverstripe-admin/issues/574
